modules/vba_documents/spec/mailers/unsuccessful_report_mailer_spec.rb

```diff
@@ -33,31 +33,59 @@
   end
 
   describe '#build' do
-    subject do
-      described_class.build(totals, [error_upload], [uploaded_upload], 7.days.ago, Time.zone.now).deliver_now
+    module MailerHelper
+      class << self
+        attr_reader :env_hash
+      end
+
+      @env_hash = {
+        staging: 'https://dsva-vetsgov-staging-benefits-documents.s3.us-gov-west-1.amazonaws.com/',
+        dev: 'https://dsva-vetsgov-dev-benefits-documents.s3.us-gov-west-1.amazonaws.com/',
+        sandbox: 'https://dsva-vagov-sandbox-benefits-documents.s3.us-gov-west-1.amazonaws.com/',
+        prod: 'https://dsva-vetsgov-prod-benefits-documents.s3.us-gov-west-1.amazonaws.com/'
+      }
     end
 
-    it 'sends the email' do
-      expect(subject.subject).to eq('Benefits Intake Unsuccessful Submission Report')
+    before do
+      current_env_url = env_url.last
+      Settings.vba_documents.location.prefix = current_env_url
+      VBADocuments::Deployment.environment = VBADocuments::Deployment.fetch_environment
+      if VBADocuments::UnsuccessfulReportMailer.const_defined?(:RECIPIENTS)
+        VBADocuments::UnsuccessfulReportMailer.send(:remove_const, :RECIPIENTS)
+      end
```

Comment:
Not certain sure why this is needed? To override any automatically set values from fetch_environment called outside the test?

Reply:
This is because the data is calculated only once during class load time. So I am, in essence, forcing a reload of the class per environment.